### ISSUE TITLE
Embedded records in old rest API have type attribute values lowercased

### ISSUE THREAD

**Issue by iammer** [emberjs/data#3726]
I am using Ember-data 2.0.0.

I have a model sent from the server using the old REST API with embedded records. It looks like this using the old rest API.  
```js
{
  parent: {
    children: [{
      type: 'SOMEVALUE'
    }]
  }
}
```
Type is an attribute of the child model.

When the child record is added to the store type is changed from 'SOMEVALUE' to 'somevalue'.

I believe this is caused by the extractRelationship method in JSONSerializer.  The relationshipHash.type is sent through Ember.string.dasherize.  Which works for the JSONAPI and for polymorphic types, but isn't the correct thing to do for embedded records in the old REST API.

**Comment by pangratz** [comment]
Hey @iammer, thanks for reporting! This is reproduced in http://ember-twiddle.com/7f0cde397742775ab97c and a fix is proposed in #3837.

**Comment by pangratz** [comment]
I think this issue can be closed since #3837 has been merged.
